### DISCUSSION TITLE
Need help with creating a scriptlet, again

### DISCUSSION THREAD

**Discussion by partingscientist** [uBlockOrigin/uBlock-issues#2514] [Q&A] [answered]
I tried writing a scriptlet to click a button whenever it shows up on Twitch.

[snipped]

The above script seemingly runs fine using the developer tools (Firefox 110), but I can't seem to make it work using uBlock Origin (1.47.3b1). Any possible reason(s) as to why? Thank you in advance.

**Comment by liamengland1** [comment]
uBO scriptlets cannot trigger any DOM events, such functionality is not implemented for security reasons. https://github.com/uBlockOrigin/uBlock-issues/issues/2229

**Reply by uBlock-user** [reply]
Nothing to do with uBO scriptlets, these scriptlets are no different that a plain javascript and that feature request is completely irrelevant here.

**Comment by uBlock-user** [comment]
replace `document.querySelector(".chat-input__buttons-container")` with `document.documentElement` and try again.

You also need to add a filter  - - `twitch.tv##+js(twitch-claim-bonus)` in my filters.

**Comment by partingscientist** [comment, marked as answer]
I think I figured it out.

I realised that uBO probably want to inject scriptlets as soon as possible to a given page. This probably means that when the above scriptlet is executed, `document.querySelector(".chat-input__buttons-container")` probably returns `null` because the DOM node has not been loaded yet. 

To fix this, I first observe the `document` on whether the `.chat-input__buttons-container` node that I'm looking for exists. Then, I disconnect that observer and reattach a new one on the node I'm targeting.

While exploring this, I also noticed that scriptlets seem to cannot have a blank line in it. I failed to find that being documented anywhere so might as well let any future reader know.

Thank you for the help!

**Reply by uBlock-user** [reply]
> While exploring this, I also noticed that scriptlets seem to cannot have a blank line in it. I failed to find that being documented anywhere so might as well let any future reader know.

Can you elaborate on this with an example ?

**Reply by mapx-** [reply]
https://forums.informaction.com/viewtopic.php?p=100840&sid=4bb87344e011911c33c5f39f91b2e99d#p100840

```
(gorhill)
Avoid blank lines inside the scriptlet code, it's used to signal end-of-scriplet. 
I use two blank lines in uBO's code just to better visually separate scriptlets.
```
